src/mustache_core.js

```diff
@@ -206,7 +206,9 @@ var core = {
 					}
 				}
 				// Look up partials in options first.
-				var partial = options.peek("partials." + localPartialName), renderer;
+				var partial = options.peek("partials." + localPartialName);
+				partial = partial || ( options.inlinePartials && options.inlinePartials[ localPartialName ] );
```

Comment:
I don't think we should warn as passing a partial on `options` is a valid way of overwriting the inline partial.  Inline partials can be thought of as defaults (similar to what's in between `<content> .... </content>`.  I like that someone can say "use this instead".